File: EllipticalRegion.java
package dymetric;

import java.awt.*;

public class EllipticalRegion extends Canvas {

    protected Color sq_colour;
    // coordinates for the ball(circle)
    protected int x_square = 10;
    protected int y_square = 150;
    protected int previous_x = x_square;
    protected int previous_y = y_square;
    protected final int sqLENGTH = 100;

    //ellipse coordinates
    protected int h = 350; // vertice
    protected int k = 175; // vertice
    protected final int a = 200; // major axis
    protected final int b = 125; // minor axis
    protected int x = h - a;
    protected int y = k - b;

    public EllipticalRegion() {
        setBackground(Color.LIGHT_GRAY); // canvas color
        sq_colour = Color.YELLOW;
    }

    // Feel free to double buffer if flickering occurs
    public void paint(Graphics g) {
        // draw an ellipse
        g.drawOval(x, y, 2*a, 2*b);

        g.clearRect(previous_x, previous_y, sqLENGTH, sqLENGTH); // erase previous square
        // draw square
        g.setColor(sq_colour);
        g.fillRect(x_square, y_square, sqLENGTH, sqLENGTH);
        previous_x = x_square;
        previous_y = y_square;
    }

    public void ellipsedSquare() {
        // condition for continuing motion
        while (x_square + sqLENGTH < 750) {
            int square_left = x_square;
            int square_right = x_square + sqLENGTH;
            int square_top = y_square;
            int square_bottom = y_square + sqLENGTH;

            // determinants for each side of the square
            int x_left_det = (int) Math.round(((double) b / a)
                    * Math.sqrt(Math.pow(a, 2) - Math.pow((square_left - h), 2)));
            int x_right_det = (int) Math.round(((double) b / a)
                    * Math.sqrt(Math.pow(a, 2) - Math.pow((square_right - h), 2)));
            int y_up_det = (int) Math.round(((double) a / b)
                    * Math.sqrt(Math.pow(b, 2) - Math.pow((square_top - k), 2)));
            int y_down_det = (int) Math.round(((double) a / b)
                    * Math.sqrt(Math.pow(b, 2) - Math.pow((square_bottom - k), 2)));

            if (square_top > k - x_left_det && square_bottom < k + x_left_det
                    && square_top > k - x_right_det && square_bottom < k + x_right_det
                    && square_left > h - y_up_det && square_right < h + y_up_det
                    && square_left > h - y_down_det && square_right < h + y_down_det) {
                sq_colour = Color.GREEN; // color for our moving body(square) inside our ellipse
            } else {
                sq_colour = Color.YELLOW; // color for our moving body(square) outside our ellipse
            }

            paint(this.getGraphics());
            
            x_square += 10;
            // introduce a delay betweeen renderings
            try {
                Thread.sleep(50);
            } catch (InterruptedException e) {
            }
        }
    }
}

File: EllipticalRegion.js
var canvas = document.getElementById("ellipse_region");
var context = canvas.getContext("2d");
context.fillStyle = "#888888"; // color for our moving body(circle)

// coordinates for the square
var x_square = 10;
var y_square = 250;
var sq_length = 100;
// draw the square
context.fillRect(x_square, y_square, sq_length, sq_length);

//elliptical coordinates
var h = 350; // vertice
var k = 225; // vertice
var a = 200; // major axis
var b = 150; // minor axis
var x = h - a;
var y = k;

//draw ellipse
for (; x < h + a; x++) {
    y = k - (b / a) * Math.sqrt(Math.pow(a, 2) - Math.pow((x - h), 2));
    context.beginPath();
    context.arc(x, y, 1, 0, 2 * Math.PI);
    context.fill();

    y = k + (b / a) * Math.sqrt(Math.pow(a, 2) - Math.pow((x - h), 2));
    context.beginPath();
    context.arc(x, y, 1, 0, 2 * Math.PI);
    context.fill();
}

function ellipsedSquare() {
    // condition for continuing motion
    if (x_square + sq_length < 600) {
        context.clearRect(x_square - 10, y_square, sq_length, sq_length); // erase previous circle

        var square_left = x_square;
        var square_right = x_square + sq_length;
        var square_top = y_square;
        var square_bottom = y_square + sq_length;

        // determinants for each side of the square
        var x_left_det = (b / a) * Math.sqrt(Math.pow(a, 2) - Math.pow((square_left - h), 2));
        var x_right_det = (b / a) * Math.sqrt(Math.pow(a, 2) - Math.pow((square_right - h), 2));
        var y_up_det = (a / b) * Math.sqrt(Math.pow(b, 2) - Math.pow((square_top - k), 2));
        var y_down_det = (a / b) * Math.sqrt(Math.pow(b, 2) - Math.pow((square_bottom - k), 2));

        if (square_top > k - x_left_det && square_bottom < k + x_left_det &&
                square_top > k - x_right_det && square_bottom < k + x_right_det &&
                square_left > h - y_up_det && square_right < h + y_up_det &&
                square_left > h - y_down_det && square_right < h + y_down_det) {
            context.fillStyle = "#00ff00"; // color for our moving body(circle)
        } else {
            context.fillStyle = "#888888"; // color for our moving body(circle)
        }

        // redraw square
        context.fillRect(x_square, y_square, sq_length, sq_length);

        x_square += 10;
        clr_obj = setTimeout(function () {
            ellipsedSquare();
        }, 100);
    }
}